format/FormatNexusJungfrauHack.py

```diff
@@ -222,14 +223,8 @@ def _detector(self):
         return self._detector_model
 
     def _beam(self, index=None):
-        if index is None:
-            index = 0
-
-        entry = self._reader.entries[0]
-        sample = entry.samples[0]
-        beam = sample.beams[0]
-
-        self._beam_model = BeamFactory(beam, index).model
+        self._beam_factory.load_model(index)
+        self._beam_model = self._beam_factory.model
```

Comment:
Can we make the `load_model()` function return the model and avoid the race condition

Reply:
nexus.py could be refactored in this way (I agree it would be better) but I wanted consistency here with the other behavior of the modules for now.  Refactoring is out of scope of this request I think.

Reply:
This change introduces the race condition.

Reply:
Addressed in 606c09559b9ba80049aaa3e27d8a8d0acf77db6a